### ISSUE TITLE
Allow Python dependency management with Pipfile

### ISSUE THREAD

**Issue by rnhurt** [awslabs/serverless-application-model#881]
**Description:**
Pipfile and pipenv are considered to be the [current recommended standard way](https://packaging.python.org/tutorials/managing-dependencies/#managing-dependencies) of managing project dependencies. Pipenv is the tool that manages your python virtual environment and dependencies according to the Pipfile.

**Expected behaviour:**
Dependencies declared in setup.py can be automatically declared in the Pipfile, reducing duplication as it currently exists.

setup.py and all requirements should declare as broad as possible ranges, so that the user can declare their own dependency versions.  Pipfile.lock can define what the exact expected versions are for reproducing builds (i.e. a dev environment or testing environment).

**Current behaviour:**
The use of `requirements.txt` forces an extra step of maintaining these files separately when working with modern Python applications.

**Comment by jfuss** [comment]
@rnhurt Are you requesting for this repo to move to Pipenv for development or something else?

**Comment by rnhurt** [comment]
> @rnhurt Are you requesting for this repo to move to Pipenv for development or something else?

No, I was suggesting that it would be useful if the `sam build` command would accept Pipfile for dependencies.  Currently, the `--manifest` parameter only accepts "requirement" files and not Pipfile files.  As I understand it, this practice has been deprecated by the Python community.  

This ticket stems from the fact that I am currently working on a project that uses Pipfile files and using AWS SAM to build/deploy it is cumbersome.  SAM forces me to build a "requirements.txt" file from my Pipfile file and keep them in sync over time.
